csci-499-sp24/AudioCard
# Shared/Removed access notifications

Added functionality for notifications pertaining to the sharing or removal of access to a card set.
Server side changes:
  Added:
    notification.js model
    route for retrieving users' notifications
  Modified:
    sharedCardset.js model to include 'revoked' in authority ENUM
    sharedCardset routes to exclude users with revoked access from collaborator list
Client side changes:
  Modified: 
    collaboratorList to update rather than delete sharedCardset entries when revoking access
    Notifications.js to differentiate between friend request notifications and sharedCardset notifications
    
To test:
Log into two accounts, grantor and grantee
Share access of a card set from grantor to grantee.
Check for a notification on the grantee, accept that notification to remove it.
On the grantor account, remove the grantee' access to the cardset
On the grantee account, observe that access to the cardset has been revoked and accept the notification.
